packages/tx/src/eip1559Transaction.ts

```diff
@@ -152,6 +154,23 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
       s,
     ] = values
 
+    const integerValues: RlpValues = {
+      nonce: nonce,
+      maxPriorityFeePerGas: maxPriorityFeePerGas,
+      maxFeePerGas: maxFeePerGas,
+      gasLimit: gasLimit,
+      value: value,
+      v: v,
+      r: r,
+      s: s,
+    }
+
+    const res = hasLeadingZeroes(integerValues)
+
+    if (res) {
+      throw new Error(`${res} cannot have leading zeroes`)
+    }
```

Comment:
We could also throw directly in `hasLeadingZeroes()` directly I guess and safe a bit of code or has this side effects? Totally no strong opinion though.

Reply:
Hmm, I guess that's a fair point and I hadn't thought about it.  Either is fine with me since the main goal was to check all fields with potentially leading zeroes in one place and then just throw an error with a specific indication of which field had the leading zeroes in it.

Reply:
Yeah that's a good idea I say throw in the function, you can highlight exactly what threw there too.

Reply:
I would also suggest moving this to ethereumjs-util under I think bytes.ts would be good and you can name it something like validateNoLeadingZeroes

Reply:
Ok, as mentioned in the chat, I am ok with that change as well.